app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt

```diff
@@ -69,7 +82,19 @@ class SwapWidgetMapper {
         )
     }
 
-    fun mapTokenB(token: SwapTokenModel, tokenAmount: BigDecimal?): SwapWidgetModel {
+    fun mapTokenBAndSaveOldFiatAmount(
+        oldWidgetModel: SwapWidgetModel,
+        token: SwapTokenModel,
+        tokenAmount: BigDecimal? = null,
+    ): SwapWidgetModel {
+        var result = mapTokenB(token, tokenAmount)
+        if (oldWidgetModel is SwapWidgetModel.Content && oldWidgetModel.fiatAmount != null) {
+            result = result.copy(fiatAmount = oldWidgetModel.fiatAmount)
+        }
+        return result
+    }
+
+    fun mapTokenB(token: SwapTokenModel, tokenAmount: BigDecimal?): SwapWidgetModel.Content {
```

Comment:
maybe we can split logic for TokenB and TokenA mapping into two mappers? Single responsibility :)

Reply:
mapper doesn't any local props or state, just methods. i don't think its really necessary